presto-hive/src/main/java/com/facebook/presto/hive/cache/HiveCachingHdfsConfiguration.java

```diff
@@ -59,12 +65,17 @@ public Configuration getConfiguration(HdfsContext context, URI uri)
         @SuppressWarnings("resource")
         Configuration config = new CachingJobConf((factoryConfig, factoryUri) -> {
             try {
+                FileSystem fileSystem = (new Path(factoryUri)).getFileSystem(hiveHdfsConfiguration.getConfiguration(context, factoryUri));
+                FileSystem dataTierFileSystem = fileSystem;
+                if (cacheConfig.isCachingEnabled() && cacheConfig.getCacheType() == ALLUXIO) {
```

Comment:
We don't need to wrap AlluxioFS into CachingFS. AlluxioFS and CachingFS are both caching FS solutions. So just return AlluxioFS should be good enough. If `getCacheType() == LOCAl_RANGE`, then we return cachingFS.

Reply:
Caching FS is the abstraction for both Alluxio and LOCAL_RANGE. The real caching solution will be determined by the datatierFileSystem. We can talk about this if it is not clear.